target/linux/mpc85xx/files/arch/powerpc/platforms/85xx/ecw7210.c

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: GPL-2.0+
+
+/*
+ * edgecore ecw7210 Board Setup
+ *
+ * Copyright (C) 2019 Scott Gilliland <zeroping@gmail.com>
+ *
+ * Based on:
+ *   mpc85xx_rdb.c:
+ *      MPC85xx RDB Board Setup
+ *      Copyright 2013 Freescale Semiconductor Inc.
+ *
+ * This program is free software; you can redistribute  it and/or modify it
+ * under  the terms of  the GNU General  Public License as published by the
+ * Free Software Foundation;  either version 2 of the  License, or (at your
+ * option) any later version.
+ */
+
+#include <linux/stddef.h>
+#include <linux/kernel.h>
+#include <linux/delay.h>
+#include <linux/interrupt.h>
+#include <linux/of_platform.h>
+
+#include <asm/time.h>
+#include <asm/machdep.h>
+#include <asm/pci-bridge.h>
+#include <mm/mmu_decl.h>
+#include <asm/prom.h>
+#include <asm/udbg.h>
+#include <asm/mpic.h>
+
+#include <sysdev/fsl_soc.h>
+#include <sysdev/fsl_pci.h>
+#include "smp.h"
+
+#include "mpc85xx.h"
+
+void __init ecw7210_pic_init(void)
+{
+	struct mpic *mpic;
+
+	mpic = mpic_alloc(NULL, 0,
+	  MPIC_BIG_ENDIAN |
+	  MPIC_SINGLE_DEST_CPU,
+	  0, 256, " OpenPIC  ");
+
+	BUG_ON(mpic == NULL);
+	mpic_init(mpic);
+
+}
+
+/*
+ * Setup the architecture
+ */
+static void __init ecw7210_setup_arch(void)
+{
+	if (ppc_md.progress)
+		ppc_md.progress("ecw7210_setup_arch()", 0);
+
+	mpc85xx_smp_init();
+
+	fsl_pci_assign_primary();
+
+        printk(KERN_INFO "ECW7210 board from Edge-core\n");
+}
+
+machine_arch_initcall(ecw7210, mpc85xx_common_publish_devices);
+
+/*
+ * Called very early, device-tree isn't unflattened
+ */
+static int __init ecw7210_probe(void)
+{
+	if (of_machine_is_compatible("edgecore,ecw7210"))
+		return 1;
+        //added so that we have a chance of working with the stock DTB found in NOR flash
+        if (of_machine_is_compatible("fsl,P1020RDB"))
```

Comment:
Is any comatibility given with a recent OpenWrt kernel and the vendor tree? If not, i would drop this here.

Reply:
It at least thinks about booting and you get a serial console. Without this, there's no chance. The vendor .dtb sits in it's own partition in NOR flash, so I had it in there for that exact combination (new kernel, old .dtb). I can pull it out though. That was another thing left over from debugging, and probably not really needed now.

Reply:
Yup, please drop this.